File: frontend/src/services/prompt.ts
/** Prompt 管理 API 客户端 — Phase C Step 8 */

import api from './api'

// ── 类型定义 ──

export interface PromptItem {
  id: number
  module: string
  version: number
  content: string
  is_active: boolean
  description: string
  created_at?: string
  updated_at?: string
}

export interface PromptModule {
  module: string
  prompts: PromptItem[]
  active_id: number | null
}

export interface PromptListResponse {
  modules: PromptModule[]
}

export interface PromptUpdateRequest {
  content?: string
  is_active?: boolean
  description?: string
}

// ── API 函数 ──

export async function fetchPrompts(): Promise<PromptListResponse> {
  const { data } = await api.get<PromptListResponse>('/api/admin/prompts')
  return data
}

export async function updatePrompt(
  promptId: number,
  body: PromptUpdateRequest,
): Promise<PromptItem> {
  const { data } = await api.put<PromptItem>(`/api/admin/prompts/${promptId}`, body)
  return data
}

export async function fetchPromptHistory(promptId: number): Promise<PromptItem[]> {
  const { data } = await api.get<PromptItem[]>(`/api/admin/prompts/${promptId}/history`)
  return data
}

// ── 模块元信息 ──

export const PROMPT_MODULE_LABELS: Record<string, string> = {
  recognition: '文物识别',
  companion: '智能伴游',
  generation: '文创生成',
  story: '故事演绎',
  recommendation: '个性化推荐',
}

export const PROMPT_MODULE_ICONS: Record<string, string> = {
  recognition: '🔍',
  companion: '🧭',
  generation: '🎨',
  story: '📖',
  recommendation: '⭐',
}

export const PROMPT_MODULE_COLORS: Record<string, string> = {
  recognition: '#1890ff',
  companion: '#52c41a',
  generation: '#fa8c16',
  story: '#722ed1',
  recommendation: '#eb2f96',
}

File: backend/app/models/prompt.py
"""Prompt 管理表 — Phase C Step 8 极简版"""

from datetime import datetime

from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime
from app.models.database import Base


class Prompt(Base):
    """系统 Prompt 版本管理

    启动时从 DB 加载 is_active=True 的版本到内存缓存。
    管理后台支持：列表查看、编辑内容、切换激活版本、查看历史。
    """
    __tablename__ = "prompts"

    id = Column(Integer, primary_key=True, autoincrement=True)
    module = Column(String(50), nullable=False, index=True, comment="模块标识: recognition/companion/generation/story/recommendation")
    version = Column(Integer, nullable=False, default=1, comment="版本号，同模块内自增")
    content = Column(Text, nullable=False, comment="Prompt 完整文本内容")
    is_active = Column(Boolean, default=False, comment="是否为当前激活版本")
    description = Column(String(200), default="", comment="版本变更描述")
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
